src/block_coordinate_algorithms.jl

```diff
@@ -0,0 +1,355 @@
+abstract type BlockCoordinateMethod end
+
+function perform_bc_updates end
+
+@enum UpdateOrder full cyclic stochastic progressive
```

Comment:
It will be preferable to use an abstract type and structs here instead of an enum, that way:
1. you can create a new update order even outside of FW
2. you can add parameters that are specific to your update order in the struct itself